state/metricsmanager.go

```diff
@@ -149,8 +224,26 @@ func (m *MetricsManager) SetGracePeriod(t time.Duration) error {
 
 // IncrementConsecutiveErrors adds 1 to the consecutive errors count.
 func (m *MetricsManager) IncrementConsecutiveErrors() error {
+	m1 := MetricsManager{
```

Comment:
Is it really necessary to create a copy of the metrics manager here?

Reply:
i didn't want to duplicate the logic..